src/ripple/app/tx/impl/CreateOffer.cpp

```diff
@@ -924,18 +1332,18 @@ CreateOffer::doApply()
 {
     // This is the ledger view that we work against. Transactions are applied
     // as we go on processing transactions.
-    Sandbox view (&ctx_.view());
+    PaymentSandbox psb (&ctx_.view());
 
     // This is a ledger with just the fees paid and any unfunded or expired
     // offers we encounter removed. It's used when handling Fill-or-Kill offers,
     // if the order isn't going to be placed, to avoid wasting the work we did.
-    Sandbox viewCancel (&ctx_.view());
+    PaymentSandbox psbCancel (&ctx_.view());
 
-    auto const result = applyGuts(view, viewCancel);
+    auto const result = applyGuts(psb, psbCancel);
```

Comment:
Why the extra `applyGuts` function? It's only called from here. I'd put the guts back into this main function.